openupgradelib/openupgrade.py

```diff
@@ -1031,13 +1032,16 @@ def migrate(no_version=False, use_env=None, uid=None, context=None):
     logging purposes.
     """
     def wrap(func):
-        def wrapped_function(cr, version):
+        @wraps(func)
+        def wrapped_function(cr, version, no_version=no_version, use_env=use_env,
+                             uid=uid, context=context):
```

Comment:
You can read global variables, but as soon as you want to assign a value, creating the local one will clash with the global one. You circumvent this by creating a local variable with the same name as the global one in the function definition already.
The `use_env` parameter has the same problem, which is why I used `use_env2` for assignments.